# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1475470531 18000
#      Sun Oct 02 23:55:31 2016 -0500
# Branch stable
# Node ID 43fe18c1129e0cd0496a55c4ecb3bfa71a9252db
# Parent  0af4e27c50e0366920bb615588e764e1b4cabcac
Use gnuplot terminal background rather than draw text-obscuring rectangle (bug #49223).

* __gnuplot_draw_figure__.m: Do not use rectangle for background, except for
  gnuplot's Qt terminal which has no 'background' option.  Do the conditional
  inside gnuplot code rather than in Octave code.

* __gnuplot_print__.m: Use figure 'color' property to set gnuplot terminal
  background color option.

diff --git a/scripts/plot/util/private/__gnuplot_draw_figure__.m b/scripts/plot/util/private/__gnuplot_draw_figure__.m
--- a/scripts/plot/util/private/__gnuplot_draw_figure__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_figure__.m
@@ -33,20 +33,21 @@ function __gnuplot_draw_figure__ (h, plo
 
     if (nkids > 0)
       fputs (plot_stream, "\nreset;\n");
       fputs (plot_stream, "set autoscale keepfix;\n");
       fputs (plot_stream, "set origin 0, 0\n");
       fputs (plot_stream, "set size 1, 1\n");
       bg = get (h, "color");
       if (isnumeric (bg))
-        fprintf (plot_stream, "set obj 1 rectangle from screen 0,0 to screen 1,1 behind fc rgb \"#%02x%02x%02x\" fs solid noborder\n", round (255 * bg));
+        fprintf (plot_stream, "if (GPVAL_TERM eq \"qt\") {set obj 1 rectangle from screen 0,0 to screen 1,1 behind fc rgb \"#%02x%02x%02x\" fs solid noborder}\n", round (255 * bg));
         bg_is_set = true;
       else
         bg_is_set = false;
+        fprintf (plot_stream, "unset obj 1");
       endif
       fg_was_set = false;
 
       for i = nkids:-1:1
         type = get (kids(i), "type");
         switch (type)
           case "axes"
             if (strcmpi (get (kids (i), "tag"), "legend"))
diff --git a/scripts/plot/util/private/__gnuplot_print__.m b/scripts/plot/util/private/__gnuplot_print__.m
--- a/scripts/plot/util/private/__gnuplot_print__.m
+++ b/scripts/plot/util/private/__gnuplot_print__.m
@@ -40,16 +40,23 @@ function opts = __gnuplot_print__ (opts)
       opts.fontsize = defaultfontsize;
     else
       opts.fontsize = axesfontsize;
     endif
   endif
   ## The axes-label and tick-label spacing is determined by
   ## the font spec given in "set terminal ..."
   gp_opts = font_spec (opts);
+  bg = get (opts.figure, "color");
+  if (isnumeric (bg))
+    gp_opts = sprintf ("%s background rgb \"#%02x%02x%02x\"",
+                       gp_opts, round (255 * bg));
+  else
+    gp_opts = sprintf ("%s nobackground", gp_opts);
+  endif
 
   pipeline = "";
 
   switch (lower (opts.devopt))
     case {"eps", "eps2", "epsc", "epsc2"}
       if (any (strcmp (opts.devopt, {"eps", "epsc"})))
         gp_opts = [gp_opts " level1"];
       endif
